pandas/core/generic.py

```diff
@@ -2590,6 +2595,8 @@ def _update_inplace(self, result, verify_is_copy=True):
 
     def add_prefix(self, prefix):
         """
+        DEPRECATED: Use ``obj.rename('prefix_{}'.format)`` or similar instead.
```

Comment:
I am a bit in doubt whether I prefer this alternative or rather `obj.rename(lambda x: "prefix_" + x)` or `obj.rename(lambda x: 'prefix_{}'.format(x))` (it's just that I think such a bare format is not that easy to understand for beginners that it is a function, but of course lambda's are also a strange beast)

BTW, you need to add `columns=` to have the correct alternative I think

Reply:
agree here, using the lambda is a bit simpler for beginners.

Reply:
Ok, done.